importers/eag_importer.py

```diff
@@ -0,0 +1,213 @@
+    def can_load_this_type(self, suffix):
+        return suffix.upper() == ".TXT"
+
+    def can_load_this_filename(self, filename):
+        return filename.upper().endswith(".EAG")
+
+    def can_load_this_header(self, header):
+        return True
+
+    def can_load_this_file(self, file_contents):
+        return True
+
+    def _load_this_file(self, data_store, path, file_object, datafile, change_id):
+        filename, ext = os.path.splitext(os.path.basename(path))
+
+        # Extract date of recording and callsign from filename
+        try:
+            date_of_recording_str, callsign = filename.split("_")
+        except ValueError:
+            self.errors.append(
+                {self.error_type: f"Error in filename - cannot extract date and callsign"}
+            )
+            return
+
+        # Get date of last Sunday, ready for timestamp calculation later
+        try:
+            last_sun_date = self.get_previous_sunday(date_of_recording_str)
+        except Exception:
+            self.errors.append(
+                {self.error_type: f"Error in filename - cannot parse date {date_of_recording_str}"}
+            )
+            return
+
+        for line_number, line in enumerate(tqdm(file_object.lines()), 1):
+            if line.text.strip().startswith("#VALUE"):
+                # Skip line
+                continue
+            tokens = line.tokens(line.WHITESPACE_DELIM)
+
+            if len(tokens) != 11:
+                self.errors.append(
+                    {
+                        self.error_type: f"Error on line {line_number}. Incorrect number of tokens: {line}"
+                    }
+                )
+                continue
+
+            # separate token strings
+            time_since_sun_ms_token = tokens[0]
+            ecef_x_token = tokens[4]
+            ecef_y_token = tokens[5]
+            ecef_z_token = tokens[6]
+            heading_token = tokens[9]
+
+            try:
+                time_since_sun_ms = float(time_since_sun_ms_token.text)
+            except ValueError:
+                self.errors.append(
+                    {
+                        self.error_type: f"Error on line {line_number}. Cannot parse time since Sunday to float: {line}"
+                    }
+                )
+                continue
+
+            # Calculate the timestamp, using the time since last Sunday and the calculated last Sunday date
+            timestamp = self.calculate_timestamp(time_since_sun_ms, last_sun_date)
+            time_since_sun_ms_token.record(self.name, "timestamp", timestamp)
+
+            # Platform is based on the callsign - the user will link this as a synonym to a defined Platform
+            platform = data_store.get_platform(platform_name=callsign, change_id=change_id,)
+
+            # Sensor Type is always fixed to GPS
+            sensor_type = data_store.add_to_sensor_types("GPS", change_id=change_id).name
+            sensor = platform.get_sensor(
```

Comment:
Leaving this field blank means that it will be requested through the resolver - so in normal operation the user will be asked to fill it in. However, the testing I did for that has thrown up another couple of issues which I'll fix in a few new commits.